ci-operator/step-registry/upi/install/azure/upi-install-azure-commands.sh

```diff
@@ -196,10 +224,11 @@ az deployment group create -g $RESOURCE_GROUP \
   --parameters baseName="$INFRA_ID"
 
 echo "Deploying 03_infra"
-az deployment group create -g $RESOURCE_GROUP \
-  --template-file "03_infra.json" \
-  --parameters privateDNSZoneName="${CLUSTER_NAME}.${BASE_DOMAIN}" \
-  --parameters baseName="$INFRA_ID"
+cmd="az deployment group create -g $RESOURCE_GROUP \
+  --template-file '03_infra.json' \
+  --parameters privateDNSZoneName='${CLUSTER_NAME}.${BASE_DOMAIN}' \
+  --parameters baseName='$INFRA_ID' ${az_deployment_optional_parameters}"
+eval "${cmd}"
```

Comment:
I tried that, but failed. when `az_deployment_optional_parameters="--parameters vnetBaseName='${vnet_basename}'"`, job failed with the following error:
```
Deploying 03_infra
ERROR: {"status":"Failed","error":{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.","details":[{"code":"BadRequest","message":"{\r\n  \"error\": {\r\n    \"code\": \"InvalidResourceReference\",\r\n    \"message\": \"Resource /subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/CI-OP-7R75TYBH-6BDE4-RG/providers/Microsoft.Network/virtualNetworks/'CI-OP-7R75TYBH-6BDE4'-VNET referenced by resource /subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/ci-op-7r75tybh-6bde4-rg/providers/Microsoft.Network/loadBalancers/ci-op-7r75tybh-6bde4-kznwd-internal-lb was not found. Please make sure that the referenced resource exists, and that both resources are in the same region.\",\r\n    \"details\": [\r\n      {\r\n        \"code\": \"NotFound\",\r\n        \"message\": \"Resource /subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/CI-OP-7R75TYBH-6BDE4-RG/providers/Microsoft.Network/virtualNetworks/'CI-OP-7R75TYBH-6BDE4'-VNET not found.\"\r\n      }\r\n    ]\r\n  }\r\n}"}]}}
```

The '' quotes is passed downed .

Reply:
Or, rather, the `'` quotes, right?

What happens if you remove the `'` 
` az_deployment_optional_parameters="--parameters vnetBaseName='${vnet_basename}'"`
->
` az_deployment_optional_parameters="--parameters vnetBaseName=${vnet_basename}"`

Reply:
I did not try that, for the az command, the `vnetBaseName` parameter should support ` ' `quotes natively,  personally I do not want to do that trick (removing `'` quotes) to avoid the issue silently. Do you think that make scenes?

Reply:
Whatever, if not using `eval` sounds better, I am okay with it, too. Fix coming soon.

Reply:
Done. @patrickdillon can you help check again.